### ISSUE TITLE
Excluding a package from checks

### ISSUE THREAD

**Issue by shamasis** [nodesecurity/nsp#95]
We are having problems with one of our important package showing up vulnerabilities and migrating it to the higher version is postponed for a while. How can we whitelist an entire package from being scanned?

**Comment by shamasis** [comment]
Until nsp has a solution, I added a workaround: https://gist.github.com/shamasis/e2969a1b9505d450586c90689061918d

**Comment by evilpacket** [comment]
@shamasis we don't have a per package whitelist but we have the ability to add exceptions for specific advisories.  https://github.com/nodesecurity/nsp#exceptions

**Comment by shamasis** [comment]
Exception causes warning suppression in all packages. This is risky compared to suppressing package wise.

**Comment by blueimp** [comment]
Here's an alternative workaround that might be a little bit simpler: https://github.com/nodesecurity/nsp/issues/59#issuecomment-237906164

**Comment by shamasis** [comment]
Fellas, this needs to be addressed. With really important repos failing NSP (tough-cookie), we should be able to exclude a version of a package from NSP in .nsprc

I had to do this workaround: https://github.com/postmanlabs/postman-collection/pull/469/files

However, the NSP integrations still fail because this is a local implementation! Only solution is to remove NSP integration. :-(

**Comment by thom-nic** [comment]
I agree this would be a nice feature to have.  In our case, several vulns are being introduced via `node-pre-gyp@0.6.36 > tar-pack@3.3.0 > debug@2.2.0` and similar.  Now as we all know node-pre-gyp is not really a runtime dependency, but it is listed as a production dep so it is present during `npm install --production`.  A desirable solution would be to e.g. ignore any advisories introduced via `node-pre-gyp`.

**Comment by neutraali** [comment]
This is an obvious hole in usability - +1 for this. I don't get why we can ignore _advisories_ and not _packages_.
